doc/decisions/frontend_keyhandling.md

```diff
@@ -0,0 +1,65 @@
+# Frontend Keybindings
+
+## Problem
+
+In the development of PermaplanT, we have identified the need for a robust keybinding system to enhance the user experience. Keybindings are utilized in various parts of the application, and we need to establish a cohesive approach for their implementation.
+
+## Constraints
+
+Our keybinding solution should adhere to the following constraints:
+
+- **Ease of Integration**: The keybinding system should be straightforward to integrate into different components of the application.
+
+- **User Configuration**: We should plan for future functionality that allows users to configure keybindings according to their preferences.
+
+## Decision / Current Solution
+
+We have decided to configure keybindings using a JSON structure that includes scopes and associated keybindings.
+
+Example:
+
+```
+{
+  "global": {
+    "Ctrl+M": "navigateToMap"
+  }
+  "planting": {
+    "Escape": "exitPlantingMode"
+  }
+}
+```
+
+Components that require keybindings can define action handlers for the configured actions. Here's an example of how this can be implemented:
```

Comment:
Please one sentence per line to easier comment on individual statements:

```suggestion
Components that require keybindings can define action handlers for the configured actions.
Here's an example of how this can be implemented:
```